### ISSUE TITLE
Tidy for within-Subject Anova 

### ISSUE THREAD

**Issue by strongway** [tidyverse/broom#95]
First, Thanks for a wonderful package. I have just updated to the latest version 0.4.0. Somehow, Tidy for within-subject ANOVA is now not working. For example, 

```
datafilename="http://personality-project.org/r/datasets/R.appendix3.data"
data.ex3=read.table(datafilename,header=T)  
aov.ex3 = aov(Recall~Valence+Error(Subject/Valence),data.ex3)
tidy(aov.ex3)
```

It gives: 

```
Error in if (attr(object$terms, "intercept")) table <- table[-1, ] : 
  argument is of length zero
```

**Comment by ndiquattro** [comment]
I'm having a similar problem where `tidy()` is only returning the interaction term of a RM ANOVA.

An example:
```
library(broom)
library(dplyr)

cw <- filter(ChickWeight, Time < 4)  # makes Time 2 levels
amod1 <- aov(weight ~ Time*Diet + Error(Chick/Time), cw)

summary(amod1)
# Error: Chick
#          Df Sum Sq Mean Sq F value Pr(>F)  
# Diet       3   62.7  20.905   2.793 0.0508 .
# Residuals 46  344.3   7.485                 
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
#
# Error: Chick:Time
#           Df Sum Sq Mean Sq F value   Pr(>F)    
# Time       1 1664.6  1664.6 344.421  < 2e-16 ***
# Time:Diet  3  100.0    33.3   6.899 0.000622 ***
# Residuals 46  222.3     4.8                     
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

tidy(amod1)
#        term df   sumsq    meansq statistic      p.value    stratum
# 1 Residuals 46 344.325  7.485326        NA           NA      Chick
# 2 Time:Diet  3 100.035 33.345000  6.899224 0.0006222752 Chick:Time
# 3 Residuals 46 222.325  4.833152        NA           NA Chick:Time
```

Additionally, if I keep more than 2 levels of Time I get the same error as @strongway .

Thanks!

**Comment by diogo-almeida** [comment]
I'm having the same problem. Has anyone found a workaround?

**Comment by IndrajeetPatil** [comment]
I am working with the latest version `0.4.4` and this issue is still there.

``` r
# loading the necessary libraries
library(tidyverse)

# converting the iris dataset to long format
iris_long <- datasets::iris %>%
  dplyr::mutate(.data = ., id = dplyr::row_number(x = Species)) %>%
  tidyr::gather(
    data = .,
    key = "condition",
    value = "value",
    Sepal.Length:Petal.Width,
    convert = TRUE,
    factor_key = TRUE
  ) %>%
  tidyr::separate(
    col = "condition",
    into = c("attribute", "measure"),
    sep = "\\.",
    convert = TRUE
  ) %>%
  tibble::as_data_frame(x = .)

# specifying the model
iris.mod <-
  stats::aov(formula = value ~ attribute * measure + Error(id / (attribute * measure)),
             data = iris_long)

# checking the summary
summary(iris.mod)
#> 
#> Error: id
#>           Df Sum Sq Mean Sq F value Pr(>F)
#> Residuals  1    264     264               
#> 
#> Error: id:attribute
#>           Df Sum Sq Mean Sq
#> attribute  1  237.2   237.2
#> 
#> Error: id:measure
#>         Df Sum Sq Mean Sq
#> measure  1   1114    1114
#> 
#> Error: id:attribute:measure
#>                   Df Sum Sq Mean Sq
#> attribute:measure  1 0.7993  0.7993
#> 
#> Error: Within
#>                    Df Sum Sq Mean Sq  F value Pr(>F)    
#> attribute           1  470.1   470.1 1445.988 <2e-16 ***
#> measure             1   57.7    57.7  177.458 <2e-16 ***
#> attribute:measure   1    1.5     1.5    4.725 0.0301 *  
#> Residuals         592  192.5     0.3                    
#> ---
#> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

# tidying the ouput
broom::tidy(iris.mod)
#> Error in if (attr(object$terms, "intercept")) table <- table[-1, ]: argument is of length zero
```

Created on 2018-04-04 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

**Comment by alexpghayes** [comment]
(Reverse) duplicate of #299, resolved in #377.
